### ISSUE TITLE
Mikrotik Support / Guide

### ISSUE THREAD

**Issue by kucingpresto** [xiahualiu/wg_gaming_installer#32]
Hi, when using it on windows, it works flawlessly but when i try to use it on mikrotik so i can expose ports on other devices im having trouble to use it properly, do you have plans to make a guide to use it on mikrotik router? Any help appreciated.

Thanks.

**Comment by xiahualiu** [comment]
Ok no problem, I am not very familiar with mikrotik, though.

You want to run the WG client on the router, and let the other devices connecting to mikrotik, to have their network traffic tunneled by the WG client on mikrotik, is my understanding correct?

**Comment by kucingpresto** [comment]
Yes, i want to do that. i was able to connect my mikrotik to the WG server, but having problem with the routing. basically after configuring nat on the mikrotik side, client arent able to receive the handshake because of the traffic replies still going through the wan instead of wg.

feel free to close the issues tho if its going to far outside of this project.

**Comment by xiahualiu** [comment]
I think the overall architecture will be very similar with or without WG on your router. 

The only difference is, after you add WG, now your routing rules should be managed by WG. This means you can control whether if a packet is proxied by WG tunnel or just pass through by filtering with the AllowedIPs field. If a packet matches the filter, it will be routed to WG, and there got tunneled. Well in reality WG just generated routing rules based on that. If you have other routing rules on the system, they may interfere with it.

Your network setup should be good. I suggest first making sure WG can add routing rules correctly, then adjust the AllowedIPs on the router to the non localhost IP ranges, to prevent looping.

https://www.reddit.com/r/WireGuard/comments/e5mm3z/how_to_allow_local_ips_on_windows_10_client/
